### PR TITLE
Added jest-puppeteer support for browser tests

### PR DESCRIPTION
**Pull request by justinboswell**
* Unified build script (install.js and build.js are no longer copy pasta)
* Added -D argument support to build script:
  * ```npm run install -- -DLibCrypto_INCLUDE_DIR=/opt/ssl/include```
* Moved tests with their impls
* Added ```test:node``` and ```test:browser``` targets. ```test``` command/target runs the node tests only by default. ```test:browser``` will install puppeteer and prereqs first.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by justinboswell** [comment]
Example run of browser tests: https://github.com/awslabs/aws-crt-nodejs/pull/90/checks?check_run_id=719021654#step:2:1500

File: lib/common/platform.spec.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import * as platform from './platform'
+
+test('platform.is_nodejs is correct', () => {
```
Comment (justinboswell) [inline lib/common/platform.spec.ts:18]: Just some basic tests that run on both platforms

File: test/native/jest.config.js [hunk 2]
```diff
@@ -11,12 +14,12 @@ module.exports = {
...
+        "^.+\\.ts?$": ['ts-jest'],
     },
     testPathIgnorePatterns: [
         '/node_modules/'
     ],
     modulePathIgnorePatterns: [
         '/cmake-js/'
     ],
-    testRunner: "jest-circus/runner"
 }
```
Comment (graebm) [inline test/native/jest.config.js:25]: trivial: lots of these files have no newline at end of file.

Comment (justinboswell) [reply test/native/jest.config.js:25]: The JS linter actually removes them

**Comment by justinboswell** [comment]
> Saw this at the end of the output. If async ops hanging past the end of a run is a wild west situation, it may not be critical, but the output may be nice to have:
> 
> > Force exiting Jest: Have you considered using `--detectOpenHandles` to detect async operations that kept running after all tests finished?
> 
> Looking at ci.yml, I'm not seeing how Windows is being excluded from the browser tests. It makes sense that they can't run on Windows though.

The browser tests only run on al2, because that's the only place npm run test:browser:ci is run.

As for the open handles thing, one day we will figure out why that happens, but jest is the only thing that ever reports it as a problem. Normal JS programs don't have any issue exiting.

**Pull request merged**